### DISCUSSION TITLE
My problem is that the SAN can't identify the correct MAC when multiple network cards are started according to the MAC

### DISCUSSION THREAD

**Discussion by wangtieying** [ipxe/ipxe#779] [Ideas]
My problem is that when multiple network cards are started, the SAN can't identify the correct MAC according to the MAC startup. For example, one wireless network card has two wired network cards. The system identifies the wireless network card as net0, while the system identifies the network card that is started by ipxe as net1. Is there any good way to start the corresponding SAN through IP or any other way to start multiple clients with one ipxe server

**Comment by NiKiZe** [comment]
I don't see how this is any issue or error in iPXE so moved to discussion instead.

As identifiers, consider using uuid or serial instead.

The booted interface should show as netX, don't use net0 or net1 in scripts unless netX has been ruled out for whatever reason.

**Reply by wangtieying** [reply]
In this case, when both the wired network card and the wireless network card exist, use the (${mac: hexhyp}) command to start the san. Only the mac of the wireless network card is recognized, but the wired network card is not

**Reply by stappersg** [reply]
> Only the mac of the wireless network card is recognized, but the wired network card is not.

So tell us more about those cards.

**Reply by wangtieying** [reply]
Maybe I said there was a problem. Both wireless network card net0 and wired network card net1 have MAC addresses. When I enable (${mac: exohyp}) to start the SAN with a mac in a wired network, the MAC address of netx, that is, the address of net0, will be displayed. Finally, it will not start

**Reply by NiKiZe** [reply]
Use netX/mac but it contains the last opened device, so unless you use ifopen net0 it should be the interface that was booted. Or the interface that succeeded on DHCP.

I repeat don't use specific net1 or similar, instead make sure you have one opened interface, that way netX will point to the correct interface, verify with the config menu.

**Comment by wangtieying** [comment]
Maybe I said there was a problem. Both wireless network card net0 and wired network card net1 have MAC addresses. When I enable (${mac: exohyp}) to start the SAN with a mac in a wired network, the MAC address of netx, that is, the address of net0, will be displayed. Finally, it will not start

**Reply by stappersg** [reply]
Take time to reply.  And more important, come with information.

At https://ipxe.org/cfg/mac is **no** _"${mac: exohyp}"_, but there is `${net0/mac:hexhyp}`


Please transform this _general github discussion_ into a _Q&A github discussion_ and ask an actual question.

**Reply by wangtieying** [reply]
Thank you very much for your help. It's already early in the morning in China. I've been studying since I got off work. I'm really sleepy, but thank you very much

**Reply by wangtieying** [reply]
It has been solved perfectly. I'm very happy

**Reply by wangtieying** [reply]
#多网卡下根据不同mac启动
:sys 
  chain /mac/boot/${/mac:hexhyp} || goto net1
:net1
  chain /mac/boot/${net1/mac:hexhyp} || goto net2
:net2
  chain /mac/boot/${net2/mac:hexhyp} || goto net3 
:net3
  chain /mac/boot/${net3/mac:hexhyp} || goto net4 
:net4
  chain /mac/boot/${net4/mac:hexhyp} || goto net5   
:net5
  chain /mac/boot/${net5/mac:hexhyp} || goto install

**Reply by stappersg** [reply]
> It has been solved perfectly. I'm very happy


Please transform this _general github discussion_ into a _Q&A github discussion_ and mark it as an answered question.


In other words: _Fine that it works for you. Make the extra effort to mark this a solved discussion._


For the larger iPXE community.
